src/components/Calculator.js

```diff
@@ -1,37 +1,50 @@
 import React from 'react';
 import './calculator.css';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = { result: 0 };
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
   }
 
+  updateCalculator = (e) => {
+    this.setState((state) => calculate(state, e.target.textContent));
+  };
+
   render() {
-    const { result } = this.state;
+    const { total, operation, next } = this.state;
     return (
       <>
-        <p className="result">{result}</p>
+        <p className="result">
+          <span>{total}</span>
+          <span>{operation}</span>
+          <span>{next}</span>
+        </p>
         <div className="calculator">
-          <button type="button" className="btn">AC</button>
-          <button type="button" className="btn">+/-</button>
-          <button type="button" className="btn">%</button>
-          <button type="button" className="orangeBtn">÷</button>
-          <button type="button" className="btn">7</button>
-          <button type="button" className="btn">8</button>
-          <button type="button" className="btn">9</button>
-          <button type="button" className="orangeBtn">x</button>
-          <button type="button" className="btn">4</button>
-          <button type="button" className="btn">5</button>
-          <button type="button" className="btn">6</button>
-          <button type="button" className="orangeBtn">-</button>
-          <button type="button" className="btn">1</button>
-          <button type="button" className="btn">2</button>
-          <button type="button" className="btn">3</button>
-          <button type="button" className="orangeBtn">+</button>
-          <button type="button" className="btn-0">0</button>
-          <button type="button" className="btn">.</button>
-          <button type="button" className="orangeBtn">=</button>
+          <button type="button" className="btn" onClick={this.updateCalculator}>AC</button>
+          <button type="button" className="btn" onClick={this.updateCalculator}>+/-</button>
+          <button type="button" className="btn" onClick={this.updateCalculator}>%</button>
+          <button type="button" className="orangeBtn" onClick={this.updateCalculator}>÷</button>
+          <button type="button" className="btn" onClick={this.updateCalculator}>7</button>
+          <button type="button" className="btn" onClick={this.updateCalculator}>8</button>
+          <button type="button" className="btn" onClick={this.updateCalculator}>9</button>
+          <button type="button" className="orangeBtn" onClick={this.updateCalculator}>x</button>
```

Comment:
[OPTIONAL]

- I think specifying an array for all button text contents and then using a map() method to display them will make your code a lot shorter and cleaner 😄